### PR TITLE
Support all component spec fields for DM

### LINKED ISSUE #4301: Support all the spec fields for DM
## Feature Request

**Is your feature request related to a problem? Please describe:**

Some new fields have been added to the custom resource specs, e.g. in https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L845, however, some of them are not supported for all the components, especially for DM, so we have to check all the supported fields in the spec and make sure they are supported by the controller:

- All the fields in https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L1937
- All the fields in https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L2054
- All the fields in https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L2119
- All the fields in https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L845

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve?

close #4301, for `ConfigUpdateStrategy`, `InitContainers`, `StatefulSetUpdateStrategy` and `PodManagementPolicy`.

### What is changed and how does it work?

add and check all fields in `ComponentSpec` can be used by DmCluster components

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support all component spec fields for DM
```

### CODE REVIEW

File: pkg/manager/member/dm_worker_member_manager.go [hunk 1]
```diff
@@ -507,9 +522,9 @@ func getNewWorkerSetForDMCluster(dc *v1alpha1.DMCluster, cm *corev1.ConfigMap) (
 				},
 			},
 			ServiceName:         controller.DMWorkerPeerMemberName(dcName),
-			PodManagementPolicy: apps.ParallelPodManagement,
+			PodManagementPolicy: baseWorkerSpec.PodManagementPolicy(),
 			UpdateStrategy: apps.StatefulSetUpdateStrategy{
-				Type: apps.RollingUpdateStatefulSetStrategyType,
+				Type: baseWorkerSpec.StatefulSetUpdateStrategy(),
```
Comment (DanielZhangQD) [inline pkg/manager/member/dm_worker_member_manager.go:527]: No need to consider the failover case here as we do for the dm-master?

Comment (csuzhangxc) [reply pkg/manager/member/dm_worker_member_manager.go:527]: for dm-worker, is it no need to use `.spec.updateStrategy.rollingUpdate.partition` feature? I don't know why and `partition` was never added for dm-worker before. and Pump without `partition` too.

Comment (DanielZhangQD) [reply pkg/manager/member/dm_worker_member_manager.go:527]: OK, I understand, there is no extra operation to do for dm-worker and Pump.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by KanShiori — APPROVED** [review]

**Pull request merged**